server.js

```diff
@@ -437,8 +439,8 @@ exports.buildExpress = function(options) {
           attachments.push(fileMeta);
         }
       });
-      attachments.concat(demo.attachments);
-      demo.attachments = attachments;
+      var oldAttachments = replacer(attachments, demo.attachments);
```

Comment:
Discussed with Ryan, we do think this is caused by the new uploader. It sends body and attachments as a multipart message, causing the demo data to end up in req.body.data instead of req.body. Bodyparser probably ignores multipart messages, so that is the reason why the extra JSON.parse is necessary.

After having given this some more thought, this only applies to /demo/update, as that is the only place where the uploader is being used.